LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -354,15 +354,9 @@ public void RegisterStudentsForClasses()
             ReportFooter(actionName: action);
         }
 
-        public void Clear()
-        {
-            // Clear all internal data structures 
-            throw new NotImplementedException();
-        }
-
         void UpdateStudentList(Classroom classroom)
         {
-            int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity - classroom.ItemList.Count + 1);
+            int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity + 1 - classroom.ItemList.Count );
```

Comment:
You don't need to change it here,  it works fine. The error in the logic. Check your **foreach** loop